state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -0,0 +1,193 @@
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+IMU Sensor Test
+* Author(s): Yashika Batra
+"""
+
+import time
+from math import sqrt
+import random
+from lib import pycubed as cubesat
+
+
+def user_test(test_type):
+    """
+    All user interaction happens in this function
+    Set wait times, prompt the users and gather IMU data and input
+    """
+
+    wait_time = 3
+
+    if test_type == "Still":
+        # print user prompts
+        print("Please leave the cubesat flat on a table. Waiting",
+              wait_time, "seconds.")
+        time.sleep(wait_time)
+        print("Collecting IMU data...")
+
+        # collect acceleration and gyroscope readings, return 0.0 mag default
+        time.sleep(wait_time)
+        acc = cubesat.acceleration()
+        gyro = cubesat.gyro()
+        print("Data Collection Complete")
+        return acc, gyro, 0.0
+
+    elif test_type == "Moving":
+        # print user prompts
+        print("Starting the moving test in", wait_time, "seconds.")
+        time.sleep(wait_time)
+        print("Please move the cubesat around for the next", wait_time,
+              "seconds, starting now.")
+        print("Collecting IMU data...")
+
+        # at some random time, collect acc and gyro readings, return 0.0 mag
+        random_wait_time = random.randint(1, wait_time)
+        time.sleep(random_wait_time)
+        acc = cubesat.acceleration()
+        gyro = cubesat.gyro()
+        time.sleep(wait_time - random_wait_time)
+        print("Data Collection Complete")
+        return acc, gyro, 0.0
+
+    elif test_type == "Rotating":
+        # print user prompts
+        print("Please rotate the cubesat as best as possible. Waiting",
+              wait_time, "seconds.")
+        time.sleep(wait_time)
+        print("Collecting IMU data...")
+
+        # at some random time, collect acc and gyro readings, return 0.0 mag
+        random_wait_time = random.randint(1, wait_time)
+        time.sleep(random_wait_time)
+        acc = cubesat.acceleration()
+        gyro = cubesat.gyro()
+        time.sleep(wait_time - random_wait_time)
+        print("Data Collection Complete")
+        return acc, gyro, 0.0
+
+    elif test_type == "Magnet":
+        # print user prompts
+        print("Please leave the cubesat flat on a table and retrieve a " +
+              "magnet. " + "Waiting", wait_time, "seconds")
+        time.sleep(wait_time)
+        print("Please slowly move the magnet closer to the cubesat for",
+              wait_time, "seconds")
+        print("Collecting IMU data...")
+
+        # collect acc, gyro, mag at the start, and mag at end
+        acc = cubesat.acceleration()
+        gyro = cubesat.gyro()
+        starting_mag = cubesat.magnetic()
+        time.sleep(wait_time)
+        ending_mag = cubesat.magnetic()
+        # calculate the change in the total magnetometer reading
+        mag = (sqrt(ending_mag[0] ** 2 + ending_mag[1] ** 2 +
```

Comment:
please clean this up using ulab. Also I don't think this is entirely accurate.
something like:
```python
from ulab.numpy.linalg import norm
mag = norm(starting_mag-ending_mag)
```
Is more accurate than what you're doing which is more like
```python
from ulab.numpy.linalg import norm
mag = norm(starting_mag)-norm(ending_mag)
```
Which wouldn't detect changes in direction at all.

Reply:
Good catch! Thanks for the suggestion. I'll also clean it up.

Reply:
Great!